### DISCUSSION TITLE
The container under test is depended on a file downloaded from a url

### DISCUSSION THREAD

**Discussion by lmilbaum** [dcermak/pytest_container#198] [Q&A]
Prior to running the container, a file needs to be downloaded. Hence, I created a fixture for downloading the file but I don't know how make the container run after it and how to setup a mount.

BTW, this is a super cool project and thank you for your help.

**Comment by dcermak** [comment]
Mounts can be setup the `volume_mounts` attribute: https://dcermak.github.io/pytest_container/usage.html#setting-up-bind-mounts-or-container-volumes

Scheduling fixtures is afaik relatively tricky with pytest. You can make one fixture request another, but that will only ensure that your download fixture runs **after** the container fixture (which you most certainly do not want). I don't know a good way how to achieve this. Instead, I would recommend to either download the file before you run your tests (assuming that it does not change frequently) or download it during container build:
```Python
ctr = DerivedContainer(base=url_to_img, containerfile="RUN curl https://url_to_file")
```

If you want a pytest only solution, you could also try to download the file before all tests run: https://stackoverflow.com/a/35394239 (but that's obviously a nogo if the files are different).

**Reply by lmilbaum** [reply]
I've chosen to download the file before all tests run. Thanks for your answer.
